libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/Moderator.java

```diff
@@ -90,7 +90,11 @@ protected T updateBasicField(U field, FieldMetaData fieldMetaData, T document, T
                                     (Map<String, Set<String>>) documentAdditions.getFieldValue(field),
                                     (Map<String, Set<String>>) documentDeletions.getFieldValue(field)));
                 } else {
-                    log.error("Unknown field in Moderator: " + field.getFieldName());
+                    try {
+                        document.setFieldValue(field, documentAdditions.getFieldValue(field));
+                    } catch (ClassCastException e) {
```

Comment:
@alexbrdn 
If you have time, I was wondering if you thought this bit of code is OK? 
I'm just adding in functionality so changes show up in moderation requests and I was unsure if using a try-catch block like this is good practice. There's an implicit cast in the call to documentAdditions.getFieldValue(field)

Reply:
I see no problem with the try-catch block per se, but don't yet understand why it is necessary. Can you explain the use-case and the problem a bit more in detail?

Reply:
Previously the code only updated the field after a moderation request is accepted if the field is a Map<String, Set<String>>. This edits it so it also updates if it's Map<String, String> - like the external Id field. I had an explicit cast in there so it said document.setFieldValue(field, (Map<String, String>)documentAdditions.getFieldValue(field)), but the IDE said it was redundant.

Reply:
I think I understand where you're coming from, but then I don't the try-catch block is necessary here at all. As I see it, just do `document.setFieldValue()` and you're good. If you want to add some special handling for the field `externalId` of `Release`, there's a good place in `ReleaseModerator` to do that. Otherwise, the one-line set should be enough, I think.

Reply:
OK, I get you, I've changed it. Thanks for your help.